pkg/entities/ie_value.go

```diff
@@ -43,2317 +42,608 @@ type InfoElementWithValue interface {
 }
 
-func (s *StringInfoElement) ResetValue() {
-	s.value = ""
+func (b *BaseInfoElement) GetLength() int {
+	return int(b.element.Len)
 }
 
-type DateTimeSecondsInfoElement struct {
-	value   uint32
-	element *InfoElement
+type Unsigned8InfoElement struct {
+	value uint8
+	BaseInfoElement
 }
 
-func NewDateTimeSecondsInfoElement(element *InfoElement, val uint32) *DateTimeSecondsInfoElement {
-	return &DateTimeSecondsInfoElement{
-		element: element,
-		value:   val,
+func NewUnsigned8InfoElement(element *InfoElement, val uint8) *Unsigned8InfoElement {
+	infoElem := &Unsigned8InfoElement{
+		value: val,
 	}
+	infoElem.element = element
+	return infoElem
 }
 
-func (dsec *DateTimeSecondsInfoElement) GetInfoElement() *InfoElement {
-	return dsec.element
-}
-
-func (dsec *DateTimeSecondsInfoElement) AddInfoElement(infoElement *InfoElement) {
-	dsec.element = infoElement
-}
-
-func (dsec *DateTimeSecondsInfoElement) GetUnsigned8Value() (uint8, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
-}
-
-func (dsec *DateTimeSecondsInfoElement) GetUnsigned16Value() (uint16, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
-}
-
-func (dsec *DateTimeSecondsInfoElement) GetUnsigned32Value() (uint32, error) {
-	return dsec.value, nil
-}
-
-func (dsec *DateTimeSecondsInfoElement) GetUnsigned64Value() (uint64, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
+func (u8 *Unsigned8InfoElement) GetUnsigned8Value() (uint8, error) {
+	return u8.value, nil
 }
 
-func (dsec *DateTimeSecondsInfoElement) GetSigned8Value() (int8, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
+func (u8 *Unsigned8InfoElement) SetUnsigned8Value(val uint8) error {
+	u8.value = val
+	return nil
 }
 
-func (dsec *DateTimeSecondsInfoElement) GetSigned16Value() (int16, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
+func (u8 *Unsigned8InfoElement) IsValueEmpty() bool {
+	return u8.value == 0
 }
 
-func (dsec *DateTimeSecondsInfoElement) GetSigned32Value() (int32, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
+func (u8 *Unsigned8InfoElement) ResetValue() {
+	u8.value = 0
 }
 
-func (dsec *DateTimeSecondsInfoElement) GetSigned64Value() (int64, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
+type Unsigned16InfoElement struct {
+	value uint16
+	BaseInfoElement
 }
 
-func (dsec *DateTimeSecondsInfoElement) GetFloat32Value() (float32, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
+func NewUnsigned16InfoElement(element *InfoElement, val uint16) *Unsigned16InfoElement {
+	infoElem := &Unsigned16InfoElement{
+		value: val,
+	}
+	infoElem.element = element
+	return infoElem
 }
 
-func (dsec *DateTimeSecondsInfoElement) GetFloat64Value() (float64, error) {
-	return 0, fmt.Errorf("accessing value of wrong data type")
+func (u16 *Unsigned16InfoElement) GetUnsigned16Value() (uint16, error) {
+	return u16.value, nil
```

Comment:
Do we still need to return error here?

Reply:
Good catch. Not needed. The rest of the code that uses these get and set functions becomes simpler.